gamemodes/terrortown/gamemode/shared/sh_sprint.lua

```diff
@@ -44,31 +44,54 @@ if SERVER then
 		ply.oldSprintProgress = ply.sprintProgress
 		ply.sprintMultiplier = bool and (1 + maxSprintMul:GetFloat()) or nil
 		ply.isSprinting = bool
-		ply.sprintTS = CurTime()
 	end)
 else
-	local function PlayerSprint(bool)
+	local lastPress = 0
+
+	local function PlayerSprint(trySprinting, bind)
 		local client = LocalPlayer()
 
-		if bool and not GetGlobalBool("ttt2_sprint_enabled", true) or not bool and not client.isSprinting then return end
+		if trySprinting and not GetGlobalBool("ttt2_sprint_enabled", true) then return end
+		if not trySprinting and not client.isSprinting or trySprinting and client.isSprinting then return end
+		if client.isSprinting and (client.useSprintBind and not bind or not client.useSprintBind and bind) then return end
 
 		client.oldSprintProgress = client.sprintProgress
-		client.sprintMultiplier = bool and (1 + GetGlobalFloat("ttt2_sprint_max", 0)) or nil
-		client.isSprinting = bool
-		client.sprintTS = CurTime()
```

Comment:
Doesn't it break something or were these the only places this var was used?

Reply:
sprintTS isn't used anywhere else or in any TTT2 related repository